### PR TITLE
Kaue article on how we do css at zumba

### PR DESCRIPTION
**Pull request by neptunz**

### CODE REVIEW

**Comment by jrbasso** [comment]
:clap: Really nice!

Few comments:
- We also keep one copy of the gist on the repo (see `_gists` folder) in case gist goes down, the gist owner accidentally delete it or the person leave the company. Can you add a copy of your gists on the same folder?
- Since this is your first article, can you add yourself on https://github.com/zumba/zumba.github.com/blob/master/_config.yml#L27 ?

**Comment by neptunz** [comment]
@jrbasso will do, thanks.

File: _posts/2015-03-31-how-we-do-css.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+---
+layout: post
+title: How We Do CSS at Zumba
+description: This is how we do CSS at Zumba.
+tags: [technology, css, sass, front end, engineering]
+author: neptunz
+---
+
+As you may or may not know Zumba's website consist of a robust app that includes a fully custom ecommerce shop that supports international transactions, class and training search and registration, Zumba Instructor Network (ZIN) admin portal and many other apps. Our in-house tech department is responsible for managing the interactive products, developing, testing and deploying the projects.
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 
> Zumba's website ~~consist~~ consists of

File: _posts/2015-03-31-how-we-do-css.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+I was inspired by the team at <a href="http://dev.ghost.org/css-at-ghost/" target="_blank">Ghost</a> and <a href="https://github.com/styleguide/css" target="_blank">Github</a> to write this article to give you an idea of how we do CSS at Zumba.
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> I was inspired by the ~~team~~ teams

File: _posts/2015-03-31-how-we-do-css.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+### Intro - Browser and Device Support
+We try to keep our code as light, modular and DRY as possible. We use Modernizr, Autoprefixer and some Foundation components. We try to keep our nesting no more than 4 levels deep and we try to make sure that  our naming convention lends itself to reusable components.
+
+We support the last few version of popular browser (Chrome, Safari and Firefox). At the time of writing this we currently support IE 8+. About 2% of our users still use IE 8. We are not out of the woods yet.
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> We support the last few ~~version~~ versions of popular ~~browser~~ browsers

Good spot for a semicolon:
> ~~About 2% of our users still use IE 8. We are not out of the woods yet.~~
> About 2% of our users still use IE 8; we are not out of the woods yet.

File: _posts/2015-03-31-how-we-do-css.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+We made the leap to a responsive application with our recent redesign of the site. Therefore, we support a wide range of desktop, tablet and mobile devices. At the time of writing this we don’t currently support smart watches. We will soon, though… so, "watch" out for that... sorry.
+
+### Preprocessor
+We use Sass with Compass as our preprocessor. Our team is proficient and has much training on this popular pre-processor and it’s large following helps.
+
+### Folder/Sandwich Structure
+Here’s the fun stuff - we structured our css based on few important factors:
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> we structured our css based ~~on few~~ on a few important factors:

File: _posts/2015-03-31-how-we-do-css.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+#### + Modularity and Reusability
+Develop components that can be reused throughout the application.
+
+#### + CSS Rules
+Consider the size of the compiled stylesheets per section of the site to not exceed IE9’s rules limit of 4,095. Our previous site was compiled into two massive stylesheets and we learned the hard way. We currently break it down per section of the site. See examples below.
+
+#### + Light stylesheets
+Keeping the stylesheet declaration low really kills to birds with one stone. In order to do that we have to break the stylesheets down to support only it’s section and that also makes our stylesheets lighter by only loading for the section it needs to support and not load unnecessary or redundant styles. Doing this helps us keep out stylesheets light.
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> really kills ~~to~~ two birds with one stone.

> Doing this helps us keep ~~out~~ our stylesheets light.

Suggested edit:
> Keeping the stylesheet declaration low really kills to birds with one stone.  ~~In order to do that we have to break the stylesheets down to support only it’s section and that also makes our stylesheets lighter by only loading for the section it needs to support and not load unnecessary or redundant styles.~~ Doing this helps us keep out stylesheets light.
> 
> Keeping the stylesheet declaration low kills two birds with one stone.  We break the stylesheets down into individual sheets that contain only the rules for each section of the site&mdash;this has the added benefit of avoiding unnecessary or redundant styles per section. Doing this helps us keep our stylesheets light.

File: _posts/2015-03-31-how-we-do-css.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+Example:
+
+`toppings
+|— _button.scss
+|— _input.scss
+|— _shares.scss
+|— _tooltip.scss`
+
+#### Sandwich
+These nom noms can contain a combination of toppings to make a complete sandwich or component. These are usually a bit more complex and a little more specific. Sandwiches can be used throughout the entire site and can be customized per section of the site to meet that sections needs or requirements.
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> can be customized per section of the site to meet that ~~sections~~ section's needs

File: _posts/2015-03-31-how-we-do-css.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+#### + Each file is named after the page or the page module it supports
+#### + We never style ID’s only Classes
+#### + Keep nesting down to a minimum and no deeper than 4 levels including pseudo selectors
+#### + Group related properties wherever possible
+#### + Media queries go after properties and the order should reflect mobile first
+
+### Naming
+
+#### Variables and Mixins
+Our variables and mixing naming is pretty specific in order to avoid any confusion. We have been here long enough to understand that the company likes to make simple to drastic changes to support the business in general. With that in mind we have to develop keeping scalability in mind and this just makes more sense for us.
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> ~~With that in mind~~ we have to develop keeping scalability in mind and this just makes more sense for us.

Suggested edit:
> We have to develop keeping scalability in mind and this variable/mixin strategy just makes more sense for us.

File: _posts/2015-03-31-how-we-do-css.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+For the most part we target based on viewport width. We have a feature or two that depends on the detection of touch. We stick to three basic breakpoints and rarely make exceptions for custom breakpoints. We don’t want to compromise code quality to consider every viewport width consideration.
+
+<script src="https://gist.github.com/neptunz/b1467ce5f11c61b22297.js"></script>
+
+#### Typography
+
+<script src="https://gist.github.com/neptunz/d2ce9353c6f1025922df.js"></script>
+
+#### Linting
+Currently our listing consists of pull request code reviews. This is what we look for:
```
Comment (young-steveo) [inline _posts/2015-03-31-how-we-do-css.md]: @neptunz 

> Currently our ~~listing~~ linting

**Pull request closed without merging**